src/main/java/com/arpnetworking/kairosdb/HistogramDataPoint.java

```diff
@@ -15,135 +15,43 @@
-        super(timestamp);
-        _precision = precision;
-        _map = map;
-        _min = min;
-        _max = max;
-        _mean = mean;
-        _sum = sum;
-    }
-
-    @Override
-    public void writeValueToBuffer(final DataOutput buffer) throws IOException {
-        buffer.writeInt(_map.size());
-        for (Map.Entry<Double, Integer> entry : _map.entrySet()) {
-            buffer.writeDouble(entry.getKey());
-            buffer.writeInt(entry.getValue());
-        }
-        buffer.writeDouble(_min);
-        buffer.writeDouble(_max);
-        buffer.writeDouble(_mean);
-        buffer.writeDouble(_sum);
-    }
-
-    @Override
-    public void writeValueToJson(final JSONWriter writer) throws JSONException {
-        writer.object().key("bins");
-        writer.object();
-        for (Map.Entry<Double, Integer> entry : _map.entrySet()) {
-            writer.key(entry.getKey().toString()).value(entry.getValue());
-        }
-        writer.endObject();
-        writer.key("min").value(_min);
-        writer.key("max").value(_max);
-        writer.key("mean").value(_mean);
-        writer.key("sum").value(_sum);
-        writer.endObject();
-    }
-
-    @Override
-    public String getApiDataType() {
-        return API_TYPE;
-    }
-
-    @Override
-    public String getDataStoreDataType() {
-        return HistogramDataPointFactory.DST;
-    }
-
-    @Override
-    public boolean isLong() {
-        return false;
-    }
-
-    @Override
-    public long getLongValue() {
-        return 0;
-    }
-
-    @Override
-    public boolean isDouble() {
-        return false;
-    }
-
-    @Override
-    public double getDoubleValue() {
-        return 0;
-    }
+    int getSampleCount();
 
     /**
-     * Gets the number of samples in the bins.
-     *
-     * @return the number of samples
+     * Getter for the sum of all samples contained in this HistogramDataPoint.
+     * @return sum of histogram samples
      */
-    public int getSampleCount() {
-        int count = 0;
-        for (Integer binSamples : _map.values()) {
-            count += binSamples;
-        }
-        return count;
-    }
-
-    public double getSum() {
-        return _sum;
-    }
+    double getSum();
 
-    public double getMin() {
-        return _min;
-    }
+    /**
+     * Getter for the min value of the samples contained in this HistogramDataPoint.
+     * @return min of histogram samples
+     */
+    double getMin();
 
-    public double getMax() {
-        return _max;
-    }
+    /**
+     * Getter for the max value of the samples contained in this HistogramDataPoint.
+     * @return max of histogram samples
+     */
+    double getMax();
 
-    public TreeMap<Double, Integer> getMap() {
-        return _map;
-    }
+    /**
+     * Getter for the map of lower bucket value to sample count contained in this HistogramDataPoint.
+     * @return map of histogram buckets
+     */
+    TreeMap<Double, Integer> getMap();
```

Comment:
Should this be a `SortedMap` or `NavigableMap` instead of the concrete `TreeMap` type?

https://docs.oracle.com/javase/8/docs/api/java/util/SortedMap.html